packages/next/src/client/components/draft-mode.ts

```diff
@@ -0,0 +1,24 @@
+import { RequestStore } from './request-async-storage'
+import { staticGenerationBailout } from './static-generation-bailout'
+
+export class DraftMode {
+  #requestStore: RequestStore
+  constructor(requestStore: RequestStore) {
+    this.#requestStore = requestStore
+  }
```

Comment:
We don't use this private method mechanism anywhere else in the codebase today. I think it's alright, but for context, you can acomplish the same effect via:

```ts
class DraftMode {
  /**
   * @internal - this declaration is stripped via `tsc --stripInternal`
   */
  private readonly requestStore: RequestStore
}
```

Reply:
That only affects TS users but JS users could still modify `private` member variables.

Thats why JS has the `#` private syntax now available since Node.js 12

Reply:
Looks like we [have a bug](https://github.com/vercel/next.js/actions/runs/4864260943/jobs/8673175422#step:6:248) in our tsconfig.json somewhere

```
/node_modules/next/dist/server/async-storage/draft-mode-provider.d.ts:9:5
Type error: Private identifiers are only available when targeting ECMAScript 2015 and higher.
```

So I changed it to use your suggestion in [65f4000](https://github.com/vercel/next.js/pull/49096/commits/65f4000ee207f0b4933897621efcf0083bd7897b)